src/main/java/com/mongodb/hibernate/internal/type/MongoStructJdbcType.java

```diff
@@ -64,10 +67,18 @@ private MongoStructJdbcType() {
 
     private MongoStructJdbcType(
             @Nullable EmbeddableMappingType embeddableMappingType, @Nullable String structTypeName) {
+        if (embeddableMappingType != null && embeddableMappingType.isPolymorphic()) {
+            throw new FeatureNotSupportedException("Polymorphic mapping is not supported");
```

Comment:
Have we decided not to support polymorphic forever for temporarily? Seems it is the latter.
Should we put ticket reference here instead?
Maybe https://jira.mongodb.org/browse/HIBERNATE-69? Or should we create a sub-task or new ticket for it?

Reply:
I mentioned in the description of that ticket that "The polymorphism should also be supported for @Embeddable @Struct." We will be able to move this part out to a new ticket if needed when we start working on polymorphims.